### ISSUE TITLE
Some Minecraft clients will timeout.

### ISSUE THREAD

**Issue by HBIDamian** [Muqsit/2FA#3]
Unfortunately, this is a client issue, and not an issue you can fix easily. 
Most people would like to log on via one device. 

The majority of the server I play on use iPhones. So, if the user needs to use the F2A App, it would timeout the MCPE client. 

Besides this, I have checked over the code briefly, and Marked it as CHECKED on poggit. 
I will test it fully tonight, to change it to Approved.

**Comment by Muqsit** [comment]
Yeah, I'm aware of this. This can be reproduced on Windows 10 edition too by _minimizing_ the game and _maximizing_ it after a period of time.

**Comment by HBIDamian** [comment]
This plugin is a great idea, but it has got problems facing it. 

- Relies on the player having multiple devices
- Relies on the player being really really quick
- Relies on external apps. (Which results in `if (!$client->focused) { $client->disconnect }`

If these were no issue at all, this plugin would be a great, secure and useful plugin.

**Comment by Muqsit** [comment]
> Relies on the player being really really quick

About that point, you can use expired codes as long as the code didn't expire 60 seconds or somewhere near that number ago. Anyway, it's just a 6 digit code, how bad could someone be in memorizing that :P
